src/main/java/com/powsybl/openloadflow/sensi/DcSensitivityAnalysis.java

```diff
@@ -822,145 +883,142 @@ public void cutConnectivity(LfNetwork lfNetwork, GraphDecrementalConnectivity<Lf
             // we will index contingencies by a list of branch that may breaks connectivity
             // for example, if in the network, loosing line L1 breaks connectivity, and loosing L2 and L3 together breaks connectivity,
             // the index would be: {L1, L2, L3}
             // todo: There may be a better way to group the contingencies that will have the same connectivities afterwards
             Collection<Contingency> nonLosingConnectivityContingencies = new LinkedList<>();
             Map<Set<ComputedContingencyElement>, List<Contingency>> contingenciesByGroupOfElementsBreakingConnectivity = new HashMap<>();
 
-            detectConnectivityLoss(lfNetwork, contingenciesStates, contingencies, contingenciesElements, equationSystem, nonLosingConnectivityContingencies, contingenciesByGroupOfElementsBreakingConnectivity);
+            detectConnectivityLoss(lfNetwork, postContingencyStates, contingencies, contingenciesElements, equationSystem,
+                    nonLosingConnectivityContingencies, contingenciesByGroupOfElementsBreakingConnectivity);
 
             Map<String, List<SensitivityValue>> contingenciesValue = new HashMap<>();
             // compute the contingencies without loss of connectivity
             for (Contingency contingency : nonLosingConnectivityContingencies) {
-                contingenciesValue.put(contingency.getId(), calculateSensitivityValues(factorGroups,
-                        factorsStates, contingenciesStates, contingency.getElements().stream().map(element -> contingenciesElements.get(element.getId())).collect(Collectors.toList())
-                ));
+                contingenciesValue.put(contingency.getId(), calculateSensitivityValues(factorGroups, factorsStates, postContingencyStates,
+                        flowStates, contingency.getElements().stream().map(element -> contingenciesElements.get(element.getId())).collect(Collectors.toList())));
             }
+
+            if (contingenciesByGroupOfElementsBreakingConnectivity.isEmpty()) {
+                return Pair.of(sensitivityValues, contingenciesValue);
+            }
+
+            GraphDecrementalConnectivity<LfBus> connectivity = lfNetwork.createDecrementalConnectivity();
+
             // compute the contingencies with loss of connectivity
             for (Map.Entry<Set<ComputedContingencyElement>, List<Contingency>> entry : contingenciesByGroupOfElementsBreakingConnectivity.entrySet()) {
                 Set<ComputedContingencyElement> breakingConnectivityCandidates = entry.getKey();
                 List<Contingency> contingencyList = entry.getValue();
-
                 lfFactors.forEach(factor -> factor.setPredefinedResult(null));
-                cutConnectivity(lfNetwork, connectivity.getConnectivity(), breakingConnectivityCandidates);
+                cutConnectivity(lfNetwork, connectivity, breakingConnectivityCandidates);
+                int mainComponent = connectivity.getComponentNumber(lfNetwork.getSlackBus());
```

Comment:
Always equal to 0

Reply:
See previous comment, I am not sure about this